packages/horizon/test/staking/delegation/addToPool.t.sol

```diff
@@ -86,5 +86,62 @@ contract HorizonStakingDelegationAddToPoolTest is HorizonStakingTest {
         staking.addToDelegationPool(users.indexer, subgraphDataServiceAddress, 1);
     }
 
-    // TODO: test recovering an invalid delegation pool
+    function test_Delegation_AddToPool_WhenInvalidPool(
+        uint256 tokens,
+        uint256 delegationTokens,
+        uint256 recoverAmount
+    ) public useIndexer useProvision(tokens, 0, 0) useDelegationSlashing() {
+        recoverAmount = bound(recoverAmount, 1, MAX_STAKING_TOKENS);
+        delegationTokens = bound(delegationTokens, MIN_DELEGATION, MAX_STAKING_TOKENS);
+
+        // create delegation pool
+        resetPrank(users.delegator);
+        _delegate(users.indexer, subgraphDataServiceAddress, delegationTokens, 0);
+
+        // slash entire provision + pool
+        resetPrank(subgraphDataServiceAddress);
+        _slash(users.indexer, subgraphDataServiceAddress, tokens + delegationTokens, 0);
+
+        // recover pool by adding tokens
+        resetPrank(users.indexer);
+        token.approve(address(staking), recoverAmount);
+        _addToDelegationPool(users.indexer, subgraphDataServiceAddress, recoverAmount);
+    }
+
+
+    function test_Delegation_AddToPool_WhenInvalidPool_RevertWhen_PoolHasNoShares(
+        uint256 tokens,
+        uint256 delegationTokens,
+        uint256 recoverAmount
+    ) public useIndexer useProvision(tokens, 0, 0) useDelegationSlashing() {
+        recoverAmount = bound(recoverAmount, 1, MAX_STAKING_TOKENS);
+        delegationTokens = bound(delegationTokens, MIN_DELEGATION, MAX_STAKING_TOKENS);
+
+        // create delegation pool
+        resetPrank(users.delegator);
+        _delegate(users.indexer, subgraphDataServiceAddress, delegationTokens, 0);
+
+        // undelegate half shares so we have some thawing shares/tokens
```

Comment:
very minor but this comment is incorrect I believe, the test is undelegating all shares

Reply:
amazing, don't know how you spotted that lol